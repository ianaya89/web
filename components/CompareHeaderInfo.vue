<template>
  <li class="compare-items" @click.stop="toggleDropdown">
    <span>
      <i class="fas fa-balance-scale" />
      <badge :value="compareItemsCount" />
    </span>
    <dropdown ref="compareDropdown" class="cursor-default" :items-classes="'top-0 -mt-6'" :activator="false" :deactivator="false">
      <ul class="compare--dropdown">
        <compare-header-info-item
          v-for="(compareItem, index) in compareItems"
          :key="index"
          :compare-item-id="compareItem"
        />
        <li class="p-4">
          <app-button class="w-full" :title="$t('app.compare.go_to_compare')" @clicked="onToCompareClick()" />
        </li>
      </ul>
    </dropdown>
  </li>
</template>

<script>
import Badge from '@/components/Badge'
import Dropdown from '@/components/Dropdown'
import CompareHeaderInfoItem from '@/components/CompareHeaderInfoItem'
import AppButton from '@/components/AppButton'

export default {
  name: 'CompareHeaderInfo',

  components: {
    AppButton,
    CompareHeaderInfoItem,
    Badge,
    Dropdown
  },

  data: () => ({
    showNotifications: false,

    notifications: {
      unreadCount: 0,
      latest: []
    }
  }),

  computed: {
    compareItems() {
      return this.$store.getters['compare/items'].map(item => Number(item))
    },

    compareItemsCount() {
      return this.compareItems.length
    }
  },

  methods: {
    toggleDropdown() {
      this.$refs.compareDropdown.toggleDropdown()
    },

    onToCompareClick() {
      this.$router.push(
        this.localePath({
          name: 'unit-id-compare',
          params: {
            id: this.$route.params.id
          },
          query: {
            items: this.compareItems.join(',')
          }
        })
      )
    }
  }
}
</script>

<style lang="postcss">
.compare-items {
  @apply .relative .list-none;

  > span {
    @apply .px-4 .h-full .flex .items-center .text-grey-d1 .font-medium .border-transparent .border-b-3 .cursor-pointer;
    &:hover {
      @apply .text-black;
    }
  }

  .compare--dropdown {
    @apply .absolute .flex .flex-col .justify-start .w-80 .max-h-128 .bg-white .shadow-md .border .border-grey-l4 .overflow-y-auto;
    right: 0;
  }
}
</style>
