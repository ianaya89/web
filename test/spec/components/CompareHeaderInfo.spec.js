import { mount } from '@vue/test-utils'
import CompareHeaderInfo from '@/components/CompareHeaderInfo'

describe('CompareHeaderInfo', () => {
  it('renders default compare header info', () => {
    const wrapper = mount(CompareHeaderInfo)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('check toggle compare dropdown', () => {
    const wrapper = mount(CompareHeaderInfo)

    wrapper.vm.toggleDropdown()
    expect(wrapper.element).toMatchSnapshot()
    wrapper.vm.toggleDropdown()
    expect(wrapper.element).toMatchSnapshot()
  })
})
