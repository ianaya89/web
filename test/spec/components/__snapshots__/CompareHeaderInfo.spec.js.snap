// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompareHeaderInfo check toggle compare dropdown 1`] = `
<li
  class="compare-items"
>
  <span>
    <i
      class="fas fa-balance-scale"
    />
     
    <div
      class="rounded-full bg-red text-white w-6 h-6 text-center text-p4 flex items-center border-white border-solid border-2"
    >
      <transition-stub
        mode="out-in"
        name="fade"
      >
        <span
          class="pt-px flex-1 font-bold leading-none"
        >
          
      0
    
        </span>
      </transition-stub>
    </div>
  </span>
   
  <div
    class="relative cursor-default"
  >
    <transition-stub
      name="fade"
    >
      <div
        class="cursor-pointer absolute top-0 left-0 right-0 bottom-0 text-center"
        style="display: none;"
      >
        <span
          class="fas fa-chevron-down text-red"
        />
      </div>
    </transition-stub>
     
    <transition-stub
      name="fade"
    >
      <div
        class="cursor-pointer absolute top-0 left-0 right-0 bottom-0 text-center"
        style="display: none;"
      >
        <span
          class="fas fa-times text-red"
        />
      </div>
    </transition-stub>
     
    <transition-stub
      name="slide-fade"
    >
      <!---->
    </transition-stub>
  </div>
</li>
`;

exports[`CompareHeaderInfo check toggle compare dropdown 2`] = `
<li
  class="compare-items"
>
  <span>
    <i
      class="fas fa-balance-scale"
    />
     
    <div
      class="rounded-full bg-red text-white w-6 h-6 text-center text-p4 flex items-center border-white border-solid border-2"
    >
      <transition-stub
        mode="out-in"
        name="fade"
      >
        <span
          class="pt-px flex-1 font-bold leading-none"
        >
          
      0
    
        </span>
      </transition-stub>
    </div>
  </span>
   
  <div
    class="relative cursor-default"
  >
    <transition-stub
      name="fade"
    >
      <div
        class="cursor-pointer absolute top-0 left-0 right-0 bottom-0 text-center"
        style="display: none;"
      >
        <span
          class="fas fa-chevron-down text-red"
        />
      </div>
    </transition-stub>
     
    <transition-stub
      name="fade"
    >
      <div
        class="cursor-pointer absolute top-0 left-0 right-0 bottom-0 text-center"
        style="display: none;"
      >
        <span
          class="fas fa-times text-red"
        />
      </div>
    </transition-stub>
     
    <transition-stub
      name="slide-fade"
    >
      <!---->
    </transition-stub>
  </div>
</li>
`;

exports[`CompareHeaderInfo renders default compare header info 1`] = `
<li
  class="compare-items"
>
  <span>
    <i
      class="fas fa-balance-scale"
    />
     
    <div
      class="rounded-full bg-red text-white w-6 h-6 text-center text-p4 flex items-center border-white border-solid border-2"
    >
      <transition-stub
        mode="out-in"
        name="fade"
      >
        <span
          class="pt-px flex-1 font-bold leading-none"
        >
          
      0
    
        </span>
      </transition-stub>
    </div>
  </span>
   
  <div
    class="relative cursor-default"
  >
    <transition-stub
      name="fade"
    >
      <div
        class="cursor-pointer absolute top-0 left-0 right-0 bottom-0 text-center"
        style="display: none;"
      >
        <span
          class="fas fa-chevron-down text-red"
        />
      </div>
    </transition-stub>
     
    <transition-stub
      name="fade"
    >
      <div
        class="cursor-pointer absolute top-0 left-0 right-0 bottom-0 text-center"
        style="display: none;"
      >
        <span
          class="fas fa-times text-red"
        />
      </div>
    </transition-stub>
     
    <transition-stub
      name="slide-fade"
    >
      <!---->
    </transition-stub>
  </div>
</li>
`;
